### ISSUE TITLE
compressed video file not play in VideoView from WebServer 

### ISSUE THREAD

**Issue by pritesh25** [Tourenathan-G5organisation/SiliCompressor#63]
i'm developing application which record the video and save to web server, and since recorded video have too much size i use compression technique SiliCompressor in order to reduce size of original video.

after applying compression technique, video size reduces and i also play that compressed video file into my VideoView and till here working perfectly.

when i upload compressed video file to my live webserver and try to play video from url in [VideoView](https://developer.android.com/reference/android/widget/VideoView), it's not playing.

However it work in my localhost server well.

can u help me what is problem ?

here is my code to play video from my live web server

  ```
VideoView vv = (VideoView)findViewById(R.id.vv);
        vv.setZOrderOnTop(true);

        try {
            MediaController mediacontroller = new MediaController(this);
            mediacontroller.setAnchorView(vv);

            Uri uri = Uri.parse(_video_link);
            vv.setMediaController(mediacontroller);
            vv.setVideoURI(uri);
            vv.seekTo(1);

        } catch (Exception e) {
            Log.e("Error", e.getMessage());
            e.printStackTrace();
        }

        vv.requestFocus();
        vv.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
            public void onPrepared(MediaPlayer mp) {

                mp.setLooping(true);
                vv.start();

            }
        });
        vv.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
            @Override
            public void onCompletion(MediaPlayer mp)
            {
                vv.resume();
            }
        });
```

**Comment by thealpha** [comment]
what are you using firebase or what as your backend

**Comment by pritesh25** [comment]
i am using php and MySQL as backend

**Comment by thealpha** [comment]
did you solved or still facing error

**Comment by thealpha** [comment]
what do you want to compress video or image

**Comment by pritesh25** [comment]
still im facing error , I'm compressing video

**Comment by thealpha** [comment]
are you compressing image

**Comment by pritesh25** [comment]
no , not now because images size is not too much if i compared with video

**Comment by thealpha** [comment]
use this to convert image it is very use full....

**Comment by pritesh25** [comment]
but of i want to compress video then what would you recommend me ?

**Comment by thealpha** [comment]
for video

**Comment by pritesh25** [comment]
yes for video

**Comment by thealpha** [comment]
wait these types of libraries does not work as you expected I will give you the code....just be calm

**Comment by pritesh25** [comment]
ok sure , I'll wait for your response.

**Comment by thealpha** [comment]
ok good !!

**Comment by thealpha** [comment]
yo !!! the file is very big how can I send you

**Comment by pritesh25** [comment]
upload the file on Google drive and share the link here or else you can mail me link personally too.

**Comment by thealpha** [comment]
do u haven facebook or email or WhatsApp like thing

**Comment by pritesh25** [comment]
yes here i my email 
priteshvishwakarma@gmail.com 
mail me on this email id

**Comment by thealpha** [comment]
okkk done I will send you now

**Comment by thealpha** [comment]
are you facing this problem
<img width="1006" alt="screen shot 2018-05-29 at 1 28 51 pm" src="https://user-images.githubusercontent.com/36503762/40774857-5048cd20-64e4-11e8-9a64-f2661035c71b.png">

**Comment by thealpha** [comment]
like your video cannot be played through video url

**Comment by pritesh25** [comment]
yes video is not playing from the URL specially when the video is compressed and uploaded to the web server

**Comment by thealpha** [comment]
how did you get that compressed file from compress video()

**Comment by pritesh25** [comment]
i use this compression technique (sillicompression)

**Comment by thealpha** [comment]
https://github.com/RudreshJR/VideoCompression/blob/master/app/src/main/java/com/rudresh/videocompression/MainActivity.java



this link helps you see this if there is problem just come here we will talk about that

**Comment by thealpha** [comment]
or
https://github.com/lalongooo/VideoCompressor 
this library by telegram

**Comment by thealpha** [comment]
SiliCompressor.with(Context).compressVideo(sourceVideoUri, destionationDirPath);


what should I pass here i am getting error ...he said that I should pass videouri, and file path but see

**Comment by thealpha** [comment]
mediaRecorder.setVideoSource(MediaRecorder.VideoSource.DEFAULT); 
mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC);
mediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
mediaRecorder.setVideoEncodingBitRate(690000);
mediaRecorder.setVideoEncoder(MediaRecorder.VideoEncoder.H264);
mediaRecorder.setVideoFrameRate(30);
mediaRecorder.setVideoSize(640, 480); this code will hep to compress even crappy but you can set quality of video

**Comment by thealpha** [comment]
why am I getting this error UrisyntexException() wtf

**Comment by pritesh25** [comment]
i already used this https://github.com/lalongooo/VideoCompressor library also but after using any compression technique to compress video file. that compressed video file not stream from the live web server (it work perfectly work in localhost).

**Comment by Tourenathan-G5organisation** [comment]
Hi @pritesh25, i don't think the problem is with SilliCompressor, because the video plays locally. Please check your file upload method. Also, try to upload the video file without compressing and see if it will play from the web server.

**Comment by pritesh25** [comment]
I know video is playing locally but it is not playing when it is uploaded on the live server.

**Comment by pritesh25** [comment]
you mean default player as a VideoView ???

**Comment by SaadShafiq** [comment]
> Hi @pritesh25, i don't think the problem is with SilliCompressor, because the video plays locally. Please check your file upload method. Also, try to upload the video file without compressing and see if it will play from the web server.

Can you guys please help me? my video is not playing locally after compression
@Tourenathan-G5organisation  @pritesh25

**Comment by Mr-Ramzan** [comment]
> > Hi @pritesh25, i don't think the problem is with SilliCompressor, because the video plays locally. Please check your file upload method. Also, try to upload the video file without compressing and see if it will play from the web server.
> 
> Can you guys please help me? my video is not playing locally after compression
> @Tourenathan-G5organisation @pritesh25

@SaadShafiq  did you resolved that issue?
Even the sample project's compressed video is not playable nither on the device not pc.

**Comment by SahilKashyap** [comment]
Hi,
Did anyone resolve this issue...?.."Video not streaming from web url after compression."

**Comment by s-komaragiri** [comment]
Does anyone have a suggested solution for this ?

**Comment by NiharDodiya** [comment]
> private Bitmap decodeFile(File f) {
>    Bitmap b = null;
>     //Decode image size
>     BitmapFactory.Options o = new BitmapFactory.Options();
>     o.inJustDecodeBounds = true;
>     FileInputStream fis = null;
>     try {
>         fis = new FileInputStream(f);
>         BitmapFactory.decodeStream(fis, null, o);
>         fis.close();
>     } catch (FileNotFoundException e) {
>         e.printStackTrace();
>     } catch (IOException e) {
>         e.printStackTrace();
>     }
>     int IMAGE_MAX_SIZE = 1024;
>     int scale = 1;
>     if (o.outHeight > IMAGE_MAX_SIZE || o.outWidth > IMAGE_MAX_SIZE) {
>         scale = (int) Math.pow(2, (int) Math.ceil(Math.log(IMAGE_MAX_SIZE /
>                 (double) Math.max(o.outHeight, o.outWidth)) / Math.log(0.5)));
>     }
>     //Decode with inSampleSize
>     BitmapFactory.Options o2 = new BitmapFactory.Options();
>     o2.inSampleSize = scale;
>     try {
>         fis = new FileInputStream(f);
>         b = BitmapFactory.decodeStream(fis, null, o2);
>         fis.close();
>     } catch (FileNotFoundException e) {
>         e.printStackTrace();
>     } catch (IOException e) {
>         e.printStackTrace();
>     }
>     Log.d(TAG, "Width :" + b.getWidth() + " Height :" + b.getHeight());
>     try {
>         FileOutputStream out = new FileOutputStream(thumb_filePath);
>         b.compress(Bitmap.CompressFormat.JPEG, 50, out);
>         out.flush();
>         out.close();
> 
>     } catch (Exception e) {
>         e.printStackTrace();
>     }
>     return b;
> }

Code working perfectly
